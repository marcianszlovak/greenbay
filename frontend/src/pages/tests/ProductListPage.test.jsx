import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import ProductListPage from '../ProductListPage';
import store from '../../store';

describe('ProductListPage Component', () => {
  const MockProductListPage = (
    <MemoryRouter>
      <Provider store={store}>
        <ProductListPage match={{ params: { id: 1 } }} />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockProductListPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
