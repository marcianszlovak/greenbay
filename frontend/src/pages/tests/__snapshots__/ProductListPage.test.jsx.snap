// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListPage Component should match the snapshot 1`] = `
Array [
  <div
    className="align-items-center row"
  >
    <div
      className="col"
    >
      <h1>
        Products
      </h1>
    </div>
    <div
      className="text-right col"
    >
      <button
        className="my-3 btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-plus"
        />
         Create Product
      </button>
    </div>
  </div>,
  <div
    className="table-responsive"
  >
    <table
      className="table-sm table table-striped table-bordered table-hover"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            NAME
          </th>
          <th>
            PRICE
          </th>
          <th>
            CATEGORY
          </th>
          <th>
            BRAND
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
]
`;
